L1Trigger/L1THGCal/test/testHGCalBackendStage1ParameterExtractor_cfg.py

```diff
@@ -59,18 +58,22 @@
 # ID of tested FPGA
 fpga_id = 1
 
+# geometry version
+from Geometry.CMSCommonData.cmsExtendedGeometry2026D49XML_cfi import XMLIdealGeometryESSource
+geometry_version = [s for s in XMLIdealGeometryESSource.geomXMLFiles.value() if "Geometry/HGCalCommonData/data/hgcal/" in s][0].replace('Geometry/HGCalCommonData/data/hgcal/','').replace('/hgcal.xml','')
```

Comment:
I would rather use regexp to match the proper string and then split the string with `/` and extract the version. Your implementation will be influenced by the order in the list and might not give the expected result.

Reply:
Now using regex to fetch `'Geometry/HGCalCommonData/data/hgcal/v.*/hgcal.xml'`. Is there any way this could still be ambiguous?

Reply:
https://github.com/hgc-tpg/cmssw/blob/17767ea36a7ed86285399c52c294d37c4e693dab/L1Trigger/L1THGCal/test/testHGCalBackendStage1ParameterExtractor_cfg.py#L65
I'm not sure to understand what the `join()` is needed for. In principle the filter should return only one match; and if there are more than one match it would mean there is an issue (in that case the best is probably to throw an exception.

Reply:
Indeed, a single match is returned, but as a list. I guess I could access the first element instead, which should be equivalent.

Reply:
Yes, it would be simpler to understand. And if the list has more than 1 element, it could throw an exception.